Xamarin.Forms.Xaml/MarkupExtensions/OnPlatformExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Globalization;
+using System.Reflection;
+
+namespace Xamarin.Forms.Xaml
+{
+	[ContentProperty("Default")]
+	public class OnPlatformExtension : IMarkupExtension
+	{
+		public object Default { get; set; }
+		public object Android { get; set; }
+		public object GTK { get; set; }
+		public object iOS { get; set; }
+		public object macOS { get; set; }
+		public object Tizen { get; set; }
+		public object UWP { get; set; }
+		public object WPF { get; set; }
+		public string Other { get; set; }
```

Comment:
Other should't be of type string, as it limits the values to be literals. Unless you plan to parse the potential markups...

I'd remove the `Other` for now, so we can think about a better solution for it (I could, e.g., modify the markup extension parser to add values in a directory instead of assigning properties)

Reply:
I thought `Other` being of type string was to allow string parsing like this.

```<Button WidthRequest="{OnPlatform Other='Foo:21,Bar25'}" />```

Reply:
Yep, indeed. So, @StephaneDelcroix removing it means we don't support custom platforms, but I guess that's fine too for a "v1"?

Reply:
Alternatively, the extension can be inherited by the user with their custom platform. You abstract away all the shared bits and then implement a new virtual method `HandlePlatform`

```
// pseudo code
public class MyOnPlatformExtension : OnPlatformExtension
{
    public object MyFooPlatform { get; set; }
    public override object HandlePlatform(object value)
    {
        if(MyFooPlatform != null && Device.RuntimePlatform == nameof(MFooPlatform))
        {
            return MyFooPlatform;
        } else {
            return base.HandlePlatform(value);
        }
    }
}
```

Reply:
Why wouldn't the `Default` be enough if all the other platforms have their own values? Effectively, the `Default` would become the value for their custom platform...

Reply:
excepting for the fact that they could potentially have more than one custom platform. Isn't that why Forms moved to stringly typed anyways?

At any rate, if you move the switch logic into a separate virtual method, we'd be able to override easily using my example above.